### PR TITLE
Spotbugs

### PR DESCRIPTION
**Pull request by efonsell**

### CODE REVIEW

File: nflow-engine/src/main/java/io/nflow/engine/config/db/MysqlDatabaseConfiguration.java [hunk 1]
```diff
@@ -45,8 +45,9 @@ public MysqlDatabaseConfiguration() {
    */
   @Bean
   @Override
-  @SuppressFBWarnings(value = { "CLI_CONSTANT_LIST_INDEX", "WEM_WEAK_EXCEPTION_MESSAGING" },
-      justification = "extracting major and minor version from splitted string, exception message is ok")
+  @SuppressFBWarnings(
+      value = { "CLI_CONSTANT_LIST_INDEX", "WEM_WEAK_EXCEPTION_MESSAGING", "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE" },
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/config/db/MysqlDatabaseConfiguration.java:49]: Is CLI_CONSTANT_LIST_INDEX required? I do not see it used here

File: nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java [hunk 2]
```diff
@@ -92,16 +93,21 @@ public String getExecutorGroupCondition() {
...
   public synchronized int getExecutorId() {
     if (executorId == -1) {
-      int hostNameMaxLength = hostMaxLength == -1
-          ? jdbc.query("select host from nflow_executor where 1 = 0", firstColumnLengthExtractor)
-          : hostMaxLength;
+      int hostNameMaxLength;
+      if (hostMaxLength == -1) {
+        hostNameMaxLength = ofNullable(jdbc.query("select host from nflow_executor where 1 = 0", firstColumnLengthExtractor))
+            .orElseThrow(() -> new IllegalStateException(
+                "Failed to resolve host name max length from database, please set correct value to nflow.executor.host.length"));
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java:100]: `Failed to resolve nflow_executor.host column max length from database`

File: nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java [hunk 3]
```diff
@@ -92,16 +93,21 @@ public String getExecutorGroupCondition() {
 
   public synchronized int getExecutorId() {
     if (executorId == -1) {
-      int hostNameMaxLength = hostMaxLength == -1
-          ? jdbc.query("select host from nflow_executor where 1 = 0", firstColumnLengthExtractor)
-          : hostMaxLength;
+      int hostNameMaxLength;
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/internal/dao/ExecutorDao.java:96]: `int hostNameMaxLength = hostMaxLength;` + lose the else?

File: nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java [hunk 4]
```diff
@@ -138,37 +141,37 @@ public WorkflowInstanceDao(SQLVariants sqlVariants, @NFlow JdbcTemplate nflowJdb
...
-    if (stateVariableValueMaxLength == -1) {
-      stateVariableValueMaxLength = jdbc.query("select state_value from nflow_workflow_state where 1 = 0",
-          firstColumnLengthExtractor);
+    return getFieldLength(stateVariableValueMaxLength, "state_value", STATE, "nflow.workflow.state.variable.value.length");
+  }
+
+  private int getFieldLength(AtomicInteger length, String field, NflowTable table, String property) {
+    int value = length.get();
+    if (value == -1) {
+      String sql = new StringBuilder("select ").append(field).append(" from ").append(table.main).append(" where 1=0").toString();
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java:169]: StringBuilder is slower than plain string concatenation here (where JVM will optimize it further).
=> Just add the strings together or even use String.format - perf does not mater.

File: nflow-engine/src/main/java/io/nflow/engine/internal/workflow/MaintenanceWorkflowStarter.java [hunk 5]
```diff
@@ -28,7 +29,7 @@
 
   protected final WorkflowInstanceFactory workflowInstanceFactory;
   protected final WorkflowInstanceService instanceService;
-  protected boolean insertOnStartup;
+  protected AtomicBoolean insertOnStartup = new AtomicBoolean();
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/internal/workflow/MaintenanceWorkflowStarter.java:32]: final

File: nflow-engine/src/main/java/io/nflow/engine/config/db/MysqlDatabaseConfiguration.java (hunk 1 above)
Comment (efonsell) [reply nflow-engine/src/main/java/io/nflow/engine/config/db/MysqlDatabaseConfiguration.java:49]: not anymore, removed

**Pull request merged**